### PR TITLE
Add basic filter to server browser

### PR DESCRIPTION
**Pull request by numas13**
Filter server list by user specified criteria (ping, map, etc). Additionally the filter send to master servers to reduce network traffic.

ps_20240902080949
ps_20240902080952
ps_20240902080959
ps_20240902081020

### CODE REVIEW

File: controls/DropDown.cpp [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+		m_scSize.h += m_scItemSize.h;
+	}
+	else
+		m_scSize.h = m_scItemSize.h;
+}
+
+void CMenuDropDown::Draw()
+{
+	uint textflags = ETF_NO_WRAP;
```
Comment (nekonomicon) [inline controls/DropDown.cpp:165]: AFAIR, `uint` is non-standart extension.
Will be better to use `unsigned` or `uint32_t`.

Comment (a1batross) [reply controls/DropDown.cpp:165]: It isn't, we define `uint` in `xash3d_types.h`

**Comment by a1batross** [comment]
I completely forgot about this PR.

Sorry!

File: controls/DropDown.cpp (hunk 1 above)
Comment (nekonomicon) [reply controls/DropDown.cpp:165]: Ok

**Pull request merged**